corehq/apps/domain/middleware.py

```diff
@@ -49,13 +47,31 @@ class DomainHistoryMiddleware(MiddlewareMixin):
 
     def process_response(self, request, response):
         if hasattr(request, 'domain') and getattr(response, '_remember_domain', True):
-            self.remember_domain_visit(request, response)
+            self.remember_domain_visit(request)
         return response
 
-    def remember_domain_visit(self, request, response):
-        last_visited_domain = request.session.get('last_visited_domain')
-        if last_visited_domain != request.domain:
-            request.session['last_visited_domain'] = request.domain
+    def remember_domain_visit(self, request):
+        if hasattr(request, 'couch_user') and request.couch_user:
+            last_visited_domain = get_last_visited_domain(request.couch_user)
+            if last_visited_domain != request.domain:
```

Comment:
This will mean that the cache doesn't get updated until it expires, so regardless of how frequently the user access HQ, the cache would be lost eventually.  I'd imagine instead that the domain cache timeout would be reset regularly while the user is accessing HQ.

An alternative would be to drop the check that it's changed and reset every time.

Reply:
dropping the check makes sense, updated in 4ecb56821ac